playground/infrastructure/test_helper.py

```diff
@@ -13,12 +13,163 @@
+def test_get_tag_when_tag_has_incorrect_format():
+    try:
+        get_tag("")
+        assert False
+    except Exception:
+        assert True
+
+
+@mock.patch('builtins.open', mock_open(read_data="data"))
+@mock.patch('helper._get_sdk')
+@mock.patch('helper._get_name')
+def test__get_example(mock_get_name, mock_get_sdk):
+    mock_get_name.return_value = "filepath"
+    mock_get_sdk.return_value = SDK_UNSPECIFIED
+
+    result = _get_example("/root/filepath.extension", "filepath.extension")
+
+    assert result == Example("filepath", "", SDK_UNSPECIFIED, "/root/filepath.extension", "data", "",
+                             STATUS_UNSPECIFIED)
+    mock_get_name.assert_called_once_with("filepath.extension")
+    mock_get_sdk.assert_called_once_with("filepath.extension")
+
+
+@mock.patch('helper._validate')
+@mock.patch('helper.get_tag')
+def test__match_pattern_with_correct_tag(mock_get_tag, mock_validate):
+    mock_get_tag.return_value = {"name": "Name"}
+
+    result = _match_pattern("/root/filepath.java")
+
+    assert result
+    mock_get_tag.assert_called_once_with("/root/filepath.java")
+    mock_validate.assert_called_once_with({"name": "Name"})
+
+
+@mock.patch('helper.get_tag')
+def test__match_pattern_without_tag(mock_get_tag):
+    mock_get_tag.return_value = None
+
+    result = _match_pattern("/root/filepath.java")
+
+    assert result is False
+    mock_get_tag.assert_called_once_with("/root/filepath.java")
+
+
+def test__match_pattern_with_unsupported_extension():
+    result = _match_pattern("/root/filepath.extension")
+
+    assert result is False
+
+
+def test__validate_without_name_field():
+    try:
+        _validate({})
+        assert False
+    except Exception:
+        assert True
+
+
+def test__validate_without_description_field():
+    try:
+        _validate({"name": "Name"})
+        assert False
+    except Exception:
+        assert True
+
+
+def test__validate_without_multifile_field():
+    try:
+        _validate({"name": "Name", "description": "Description"})
+        assert False
+    except Exception:
+        assert True
+
+
+def test__validate_with_incorrect_multifile_field():
+    try:
+        _validate({"name": "Name", "description": "Description", "multifile": "Multifile"})
+        assert False
+    except Exception:
+        assert True
+
+
+def test__validate_without_categories_field():
+    try:
+        _validate({"name": "Name", "description": "Description", "multifile": "true"})
+        assert False
+    except Exception:
+        assert True
+
+
+def test__validate_with_all_fields():
+    _validate({"name": "Name", "description": "Description", "multifile": "true", "categories": []})
+    assert True
+
+
+def test__get_name():
+    result = _get_name("filepath.extension")
+
+    assert result == "filepath"
+
+
+def test__get_sdk_with_supported_extension():
```

Comment:
Could you please check here all our supported extensions along with Java?

Reply:
Changed.